manifests/nexus.pp

```diff
@@ -65,9 +66,20 @@
 
   }.filter |$keys, $values| { $values != undef }
 
-  $artifact_url = assemble_nexus_url($url, $query_params)
-  $checksum_url = regsubst($artifact_url, "p=${packaging}", "p=${packaging}.${checksum_type}")
-
+  if $use_nexus3_urls {
+    if $classifier {
+      $c = "-${classifier}"
+    } else {
+      $c = ''
+    }
+    $artifact_url = sprintf('%s/repository/%s/%s/%s/%s/%s-%s%s.%s', $url,
```

Comment:
A slash should never be in any of the parameters. Either way, if there is a slash and it's escaped you'd get a 404 and if it's not escaped, you'd also get a 404. 

But if you're being silly and you specify the group_id with slashes (nl/avisi/foo instead of nl.avisi.foo) - escaping the slash will actually break the url.